Server/Components/Pages/GamePage.razor

```diff
@@ -246,17 +253,45 @@
 	private readonly ObservableCollection<Item> _equipment = new();
 	private readonly ObservableCollection<Item> _inventory = new();
 
-	private Dictionary<string, object> _equipmentPopupParameters = null!;
+    private User _user = new();
+	private string _username = String.Empty;
+
+    private Dictionary<string, object> _equipmentPopupParameters = null!;
 
-    private PopupContainer _popup = null!;
+	private PopupContainer _popup = null!;
+	private string _loginInfo = string.Empty;
 
 	protected override Task OnInitializedAsync()
 	{
-	    _equipmentPopupParameters = new()
+		_equipmentPopupParameters = new()
+	    {
+	        {"Inventory", _inventory},
+	        {"Equipment", _equipment},
+	        {"UserData", _user}
+		};
+		return base.OnInitializedAsync();
+	}
+
+	private async Task HandleUserLogin()
+	{
+		var user = await UsersRepository.GetUserAsync(_username);
+		if (user == null)
+		{
+			user = new User
+			{
+			    Username = _username,
+			    CreatedAt = DateTime.UtcNow
+			};
+			await UsersRepository.AddUserAsync(user);
+		}
+	    _user = user;
+		_equipmentPopupParameters["UserData"] = _user;
+	    _loginInfo = $"Zalogowano: {_user.Username}";
+        if (_popup.Component is not null && _popup.Component.Instance is not null && _popup.Component.Type == typeof(EquipmentPopup))
 	    {
-	        {"inventory", _inventory},
-	        {"equipment", _equipment}
-	    };
-	    return base.OnInitializedAsync();
+	        var equipmentPopup = (EquipmentPopup)_popup.Component.Instance;
+	        equipmentPopup.UserData = _user;
+			equipmentPopup.Refresh();
```

Comment:
I wish it wasn't this difficult to update a child component with new data (have tried for too long to find something nicer)